File: auth-service/src/main/java/com/secure_auth/auth_service/repository/UserRepository.java
package com.secure_auth.auth_service.repository;

import com.secure_auth.auth_service.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {

    Optional<User> findByUsername(String username);

    Boolean existsByEmail(String email);

    Optional<User> findByUserNameOrByEmail(String username, String email);

    boolean existsByUserName(String username);
}

File: security-auth/src/main/kotlin/com/secure_auth/security_auth/repository/UserRepository.kt
package com.secure_auth.security_auth.repository

import com.secure_auth.security_auth.entity.Users
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
public interface UserRepository: JpaRepository<Users, Long> {

    fun findByUsername(username: String): Optional<Users>

    fun existsByEmail(email: String): Boolean

    fun findByUsernameOrEmail(username: String, email: String): Optional<Users>

    fun existsByUsername(username: String): Boolean
}